pybossa/repositories/user_repository.py

```diff
@@ -54,6 +54,9 @@ def get_by(self, **attributes):
     def get_all(self):
         return self.db.session.query(User).filter_by(restrict=False).all()
 
+    def get_enbled_users_by_email(self, emails):
+        return self.db.session.query(User).filter_by(enabled=True).filter(User.email_addr.in_(emails)).all()
+
```

Comment:
Deep, just to double check here. Did you examine the SQL generated by SQLAlchemy? There might be a common problem for ORMs: n+1 select issue, where it uses loop for the query.

Reply:
confirmed w/ debug that there's single query generated
```python
(Pdb++) orms = self.db.session.query(User).filter_by(enabled=True).filter(User.email_addr.in_(emails))
(Pdb++) str(orms)
'SELECT "user".id AS user_id, "user".created AS user_created, "user".email_addr AS user_email_addr, "user".name AS user_name, "user".fullname AS user_fullname, "user".locale AS user_locale, "user".api_key AS user_api_key, "user".passwd_hash AS user_passwd_hash, "user".ldap AS user_ldap, "user".admin AS user_admin, "user".pro AS user_pro, "user".privacy_mode AS user_privacy_mode, "user".restrict AS user_restrict, "user".category AS user_category, "user".flags AS user_flags, "user".twitter_user_id AS user_twitter_user_id, "user".facebook_user_id AS user_facebook_user_id, "user".google_user_id AS user_google_user_id, "user".ckan_api AS user_ckan_api, "user".newsletter_prompted AS user_newsletter_prompted, "user".valid_email AS user_valid_email, "user".confirmation_email_sent AS user_confirmation_email_sent, "user".subscribed AS user_subscribed, "user".consent AS user_consent, "user".info AS user_info, "user".subadmin AS user_subadmin, "user".enabled AS user_enabled, "user".user_pref AS user_user_pref, "user".last_login AS user_last_login \nFROM "user" \nWHERE "user".enabled = true AND "user".email_addr IN ([POSTCOMPILE_email_addr_1])'
```

Reply:
That's great. Thanks for confirming.